ui/src/shared/components/CloudUpgradeButton.tsx

```diff
@@ -1,25 +1,56 @@
 // Libraries
 import React, {FC} from 'react'
 import {Link} from 'react-router'
+import {connect} from 'react-redux'
 
 // Components
 import CloudOnly from 'src/shared/components/cloud/CloudOnly'
 
 // Constants
 import {CLOUD_URL, CLOUD_CHECKOUT_PATH} from 'src/shared/constants'
+import {
+  HIDE_UPGRADE_CTA_KEY,
+  PAID_ORG_HIDE_UPGRADE_SETTING,
+} from 'src/cloud/constants'
 
-const CloudUpgradeButton: FC = () => {
+// Types
+import {AppState, OrgSetting} from 'src/types'
+
+interface StateProps {
+  show: boolean
+}
+
+const CloudUpgradeButton: FC<StateProps> = ({show}) => {
   return (
     <CloudOnly>
-      <Link
-        className="cf-button cf-button-sm cf-button-success upgrade-payg--button"
-        to={`${CLOUD_URL}${CLOUD_CHECKOUT_PATH}`}
-        target="_self"
-      >
-        Upgrade Now
-      </Link>
+      {show ? (
+        <Link
+          className="cf-button cf-button-sm cf-button-success upgrade-payg--button"
+          to={`${CLOUD_URL}${CLOUD_CHECKOUT_PATH}`}
+          target="_self"
+        >
+          Upgrade Now
+        </Link>
+      ) : null}
     </CloudOnly>
   )
 }
 
-export default CloudUpgradeButton
+const mstp = ({
+  cloud: {
+    orgSettings: {settings},
+  },
+}: AppState) => {
+  const hideUpgradeCTA = settings.find(
```

Comment:
`hideUpgradeCTA` is this an object? like an object of settings or a boolean? the name kind of reads like a function. i think thinking about the name of this might make the intent a little clearer.